dcpy/utils/versions.py

```diff
@@ -157,27 +174,80 @@ def parse(v: str) -> Version:
             return MajorMinor(year=int(m[1]), major=int(m[2]))
         case r"^(\d{2})v(\d+)\.(\d+)$" as m:
             return MajorMinor(year=int(m[1]), major=int(m[2]), minor=int(m[3]))
+        case r"^(\d{2})v(\d+)\.(\d+)\.(\d+)$" as m:
+            return MajorMinor(
+                year=int(m[1]), major=int(m[2]), minor=int(m[3]), patch=int(m[4])
+            )
         case r"^(\d{2})Q(\d)$" as m:
             return Date(
                 date(2000 + int(m[1]), int(m[2]) * 3 - 2, 1),
                 format=DateVersionFormat.quarter,
             )
+        case r"^(\d{2})Q(\d)\.(\d+)$" as m:
+            return Date(
+                date(2000 + int(m[1]), int(m[2]) * 3 - 2, 1),
+                format=DateVersionFormat.quarter,
+                patch=int(m[3]),
+            )
         case r"^(\d{4})-(\d{2})-(\d{2})$" as m:
             return Date(
                 date(int(m[1]), int(m[2]), int(m[3])),
                 format=DateVersionFormat.date,
             )
+        case r"^(\d{4})-(\d{2})-(\d{2})\.(\d+)$" as m:
+            return Date(
+                date(int(m[1]), int(m[2]), int(m[3])),
+                format=DateVersionFormat.date,
+                patch=int(m[4]),
+            )
         case r"^(\d{4})-(\d{2})$" as m:
             return Date(
                 date(int(m[1]), int(m[2]), 1),
                 format=DateVersionFormat.month,
             )
+        case r"^(\d{4})-(\d{2}).(\d+)$" as m:
+            return Date(
+                date(int(m[1]), int(m[2]), 1),
+                format=DateVersionFormat.month,
+                patch=int(m[3]),
+            )
         case _:
             raise ValueError(
                 f"Tried to parse version {v} but it did not match the expected format"
             )
 
 
+def sort(versions: list[Version]) -> list[Version]:
+    """
+    Sorts version in ascending order: from oldest to newest.
+    """
+    version_types = set([type(v) for v in versions])
+    # can't compare different types
+    if len(version_types) != 1:
+        # only handle date-like versions
+        raise TypeError(
+            f"Can't sort mixed types of dataset versions: {[v.__name__ for v in version_types]}"
+        )
+    return sorted(versions)
+
+
+def is_newer(version_1: str, version_2: str) -> bool:
+    """
+    Compares `version_1` to `version_2`. Returns True if `version_1` is newer than `version_2`.
+    Both versions are expected to be of same Version subtype.
+    """
+    try:
+        version_1_obj = parse(version_1)
+        version_2_obj = parse(version_2)
+        versions_sorted = sort([version_1_obj, version_2_obj])
```

Comment:
nit - can just return `version_1_obj > version_2_obj` here, no?

This is also maybe a sign that we should just use `Version` objects and serialize to strs rather than deal with strs throughout the codebase... but again lets get our version logic a little more mature

Reply:
makes sense!